src/build/smart-contracts.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Smart Contracts
+---
+
+# Smart Contracts
+
+This article explains the Deployment and interaction of CW20.
+
+### Install rustup
+
+First, install [rustup](https://rustup.rs/). Once installed, make sure you have the wasm32 target. This is an important step before running any contract to ensure that you are running Rust 1.51.0+ with wasm32-unknown-unknown target installed.
+
+```
+rustup --version
+rustup default stable
+cargo --version
+rustup target list --installed
+```
+
+If the rust version is less than 1.51.0+, update all dependencies in the lockfile for cargo and rustup by running the command:
+```
+cargo update
+rustup update
+```
+
+## Deployment and interaction of CW20
+
+[CW20](https://github.com/CosmWasm/cw-plus/tree/v0.9.0/contracts/cw20-base) is equivalent to ERC20. CW-20 is similar, in some respects, to bitcoin, Litecoin, and any other cryptocurrency. CW-20 tokens are blockchain-based assets that have value and can be sent and received. The primary difference is that instead of running on their own blockchain, CW-20 tokens are issued on the CosmWasm network.
+
+This section explains how to deploy and interact with CW20 using cudos-noded.
+
+### Functions present in CW20 base standard
+
+The specification of CW20 are represented within the following:
+* [Base](https://github.com/CosmWasm/cw-plus/blob/v0.9.0/packages/cw20/README.md#base)
+* [Allowances](https://github.com/CosmWasm/cw-plus/blob/v0.9.0/packages/cw20/README.md#allowances)
+* [Mintable](https://github.com/CosmWasm/cw-plus/blob/v0.9.0/packages/cw20/README.md#mintable)
+
+Note that you can find all messages, actions, and queries within the list above.
+
+### Get the binaries and download the CosmWasm Plus contracts
+
+* Follow the [instructions to start and build the binaries](/build/start-binaries.html)
+* Clone the [cw-plus](https://github.com/CosmWasm/cw-plus) repo with [release tag of v0.9.0](https://github.com/CosmWasm/cw-plus/tree/v0.9.0) by running the command:
+```
+git clone --depth 1 --branch v0.9.0 https://github.com/CosmWasm/cw-plus.git
+```
+
+### Compile the contracts
+
+- Navigate to cw-plus folder and run the following command:
+
+```
+foo@bar:~$ cd cw-plus
+foo@bar:~$ docker run --rm -v "$(pwd)":/code \
+--mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+--mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+cosmwasm/workspace-optimizer:0.12.3
+```
+
+Note that compilation will take some time and will provide you with artifacts subfolder. If you hit any issues there and want to debug, you can try to run the following in the contract dir:
```

Comment:
according to https://github.com/CosmWasm/cw-plus/tree/v0.9.0#compiling, it should be:
>If you hit any issues there and want to debug, you can try to run the following in ***each*** contract dir

but i think that sentence and the line below can probably be removed as in that it can cause more confusion than solutions